### PR TITLE
Fuse status for OSX; Launchd status fixes

### PR DESCRIPTION
**Pull request by gabriel**

### CODE REVIEW

**Comment by maxtaco** [comment]
@gabriel let's do a hang out with @patrickxb today to go over this PR?  I'm curious to learn more about the internals (I'm guessing @patrickxb is too!)

**Comment by patrickxb** [comment]
Sounds good.  I'm free whenever.

**Comment by gabriel** [comment]
yeah whenever

File: go/client/fuse_status_osx.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	status := keybase1.FuseStatus{
+		BundleVersion: bundleVersion,
+		InstallStatus: keybase1.InstallStatus_UNKNOWN,
+		InstallAction: keybase1.InstallAction_UNKNOWN,
+	}
+
+	var kextInfo *kext.Info
+
+	// Check osxfuse 2.x
+	path2 := "/Library/Filesystems/osxfusefs.fs"
```
Comment (maxtaco) [inline go/client/fuse_status_osx.go]: can these 2 blocks share code?  Seems like a lot of code duplication between the `2.x` and `3.x` cases.

File: go/client/fuse_status_osx.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			status.Error = &keybase1.StatusError{
+				Message: fmt.Sprintf("Fuse (2) installed (%s) but kext was not loaded (%s)", status.Path, kextID2)}
+			return status
+		}
+		// Installed (v2)
+		status.KextID = kextID2
+		kextInfo = infov2
+	}
+
+	// Check osxfuse 3.x
```
Comment (maxtaco) [inline go/client/fuse_status_osx.go]: see note above about code sharing

File: go/protocol/keybase_v1.go [hunk 3]
```diff
@@ -367,7 +364,7 @@ const (
 	InstallStatus_INSTALLED     InstallStatus = 4
 )
 
-type ServiceStatusError struct {
+type StatusError struct {
```
Comment (maxtaco) [inline go/protocol/keybase_v1.go]: Is it possible to use our regular `Status` field here? We're using that elsewhere to move Go Errors over msgpack RPC.

File: go/client/cmd_fuse_osx.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	}
+}
+
+func NewCmdFuseStatus(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	return cli.Command{
+		Name:         "status",
+		ArgumentHelp: "<kext-label> <bundle-version>",
+		Usage:        "Status for fuse, including for installing or updating",
+		Action: func(c *cli.Context) {
+			g.Env.SetSkipLogForward()
```
Comment (maxtaco) [inline go/client/cmd_fuse_osx.go]: You should rebase, I changed the way this is handled...

File: go/client/fuse_status_osx.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	status := keybase1.FuseStatus{
+		BundleVersion: bundleVersion,
+		InstallStatus: keybase1.InstallStatus_UNKNOWN,
+		InstallAction: keybase1.InstallAction_UNKNOWN,
+	}
+
+	var kextInfo *kext.Info
+
+	// Check osxfuse 3.x
+	path3 := "/Library/Filesystems/osxfuse.fs"
```
Comment (maxtaco) [inline go/client/fuse_status_osx.go]: My comment was blown away, but did you see it earlier, that this block and the subsequent block can be factored into one? Seems like a lot of code repetition.

Comment (gabriel) [inline go/client/fuse_status_osx.go]: yeah I can move into a method and re-use

File: go/protocol/keybase_v1.go (hunk 3 above)
Comment (gabriel) [inline go/protocol/keybase_v1.go]: yeah

File: go/client/fuse_status_osx.go (hunk 5 above)
Comment (gabriel) [inline go/client/fuse_status_osx.go]: I started re-factoring this and it became harder to read/process... It's filling in stuff and then may error and return at any moment. I think it's probably ok to be repetitive here cause it's super obvious whats happening?

Comment (maxtaco) [inline go/client/fuse_status_osx.go]: ok!

**Pull request merged**